### DISCUSSION TITLE
Moving forward as new maintainers

### DISCUSSION THREAD

**Discussion by miguel-a-calles-mba** [serverless-heaven/serverless-webpack#1544] [Team Posts]
@HyperBrain added new maintainers, of which I am one. I thought I'd start a discussion to determine how we move forward.

Does anybody have suggestions on how we review and approve pull requests?

How should we test a release candidate and create a new release?

How do we update npm?

Thanks,
Miguel

**Comment by rdsedmundo** [comment]
Hello Miguel.

Since most of us aren't too much familiar with the codebase yet, I'd suggest that we do at least 2 reviews per pull request, at least at the beginning.

Regarding NPM, from what I understand from the message HyperBrain sent to you, he'll be still responsible for publishing the NPM package when we merge something new and give him the green light.

I'd suggest for now we focus on those critical bug fixes on the PRs opened and then we can think about a new version. What do you think?

**Comment by HyperBrain** [comment]
Hi guys,

yes, I will then release on NPM as soon as a new one is ready.
Regarding how I did the releases, you can check the release commits I did in the past (the ones which are tagged properly with the version).

PR build settings:
Currently the PRs are build and checked with a Travis build before they can be merged. Additionally, the code coverage of the unit tests is checked and must not decrease with a PR, i.e. for every code committed there have to be unit tests. Also, at least one approval is needed. Only then it is possible to merge.
In the past this proved to get a highly stable product, without many regressions.

And a big thank you to you for helping out here 👍 🥇 

cu
Frank

**Comment by miguel-a-calles-mba** [comment]
How should we increment the package version?

Do we do it after each PR?

Do we start a "release" PR?

Some other approaches?

Thoughts?

**Comment by miguel-a-calles-mba** [comment]
How do we address PRs that cannot build?

Several PRs have this status.

![image](https://user-images.githubusercontent.com/44813512/79128635-b5dd6180-7d58-11ea-8631-1575ea763a68.png)

I just created a PR to address npm security vulnerabilities and it cannot build either.

Thoughts?

**Comment by miguel-a-calles-mba** [comment]
> How should we increment the package version?
> 
> Do we do it after each PR?
> 
> Do we start a "release" PR?
> 
> Some other approaches?
> 
> Thoughts?

Just saw this.

```
Releasing a new version

Create a new milestone with the intended version number, and assign any relevant
issues and/or pull requests to it. Pull requests should be merged only if all
checks pass and the code coverage thresholds are met. This usually means that
contributors will often have to write tests as part of the submission process.
```

**Comment by jamesmbourne** [comment]
@miguel-a-calles-mba it seems like those builds on PRs created by Dependabot are running in Travis CI, but the status does not seem to be getting reported back correctly:

![image](https://user-images.githubusercontent.com/594697/79158904-0db6af80-7dcf-11ea-94e6-c8f9e82cf6bb.png)

https://travis-ci.org/github/serverless-heaven/serverless-webpack/builds/674208170

**Comment by miguel-a-calles-mba** [comment]
@jamesmbourne That's great. How do you find them in Travis CI?

**Comment by miguel-a-calles-mba** [comment]
@HyperBrain, how can we fix the Travis CI integration?

**Comment by jamesmbourne** [comment]
@miguel-a-calles-mba I was just looking at them on the branches list here https://travis-ci.org/github/serverless-heaven/serverless-webpack/branches.

I did find this issue on the Travis forums which sounds similar: https://travis-ci.community/t/known-issue-travis-ci-reports-expected-waiting-for-status-to-be-reported-on-the-github-status-api-but-the-status-never-arrives/1154, some of the suggestions are to re-authorize Travis with GitHub, but I don't think I've got sufficient access to do that.

**Comment by HyperBrain** [comment]
I will try to reauth it later today

**Comment by HyperBrain** [comment]
The new GitHub App is only supported by travis-ci.com, not travis-ci.org anymore. I started a migration of the builds to there (they state OpenSource builds are completely free). As soon as everything is migrated, it should work again. Right now the new App notification targets are not yet there.... I will wait for some time to check if it is there again.

**Comment by miguel-a-calles-mba** [comment]
Thanks, @HyperBrain. It's been working consistently now.
